src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ViewComponentTagHelperDescriptorFactory.cs

```diff
@@ -204,6 +205,20 @@ private void AddBoundAttributes(ImmutableArray<IParameterSymbol> methodParameter
                     attributeBuilder.DisplayName = $"{simpleName} {containingDisplayName}.{parameter.Name}";
                     attributeBuilder.SetPropertyName(parameter.Name);
 
+                    if (parameter.HasExplicitDefaultValue)
+                    {
+                        // Respect `null` being pass as the default value for a parameter, but only if the
+                        // parameter is decorated with a nullable annotation.
+                        if (parameter.ExplicitDefaultValue is null && parameter.NullableAnnotation is NullableAnnotation.Annotated)
+                        {
+                            attributeBuilder.DefaultValue = "null";
+                        }
+                        else if (parameter.ExplicitDefaultValue is not null)
+                        {
+                            attributeBuilder.DefaultValue = parameter.ExplicitDefaultValue.ToString();
```

Comment:
Hmm, does this handle: 
![image](https://user-images.githubusercontent.com/2008729/119576027-d4d86080-bd6c-11eb-97f5-33ac9b6da57d.png)